Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixSearchId.tsx

```diff
@@ -0,0 +1,280 @@
+                    placeholder="Enter Computer Company"
+                />
+            );
+        }
+    };
+
+    private _renderNameTable = () => {
+        if (!this.state.isFetched) {
+            return null;
+        }
+
+        if (!this.state.isValidSearch) {
+            return (
+                <p className="text-center text-danger">
+                    No computer found by this name, please try again.
+                </p>
+            );
+        }
+        return (
+            <Table>
+                <tbody>
+                    {this.state.listOfComputers.map(computer => {
+                        return (
+                            <tr
+                                className="bigfix-info"
+                                onClick={() =>
+                                    this.props.addBigFixId("systemManagementId", computer.id)
+                                }
+                                key={computer.id}
+                            >
+                                <td>{computer.name}</td>
+                            </tr>
+                        );
+                    })}
+                </tbody>
+            </Table>
+        );
+    };
+
+    private _renderSearchButton = () => {
+        if (this.state.isSearching) {
+            return (
+                <Button color="primary" disabled={false}>
+                    {" "}
+                    <i className="fas fa-lg fa-spinner fa-pulse" />
+                </Button>
+            );
+        } else {
+            return (
+                <Button
+                    color="primary"
+                    disabled={false}
+                    onClick={() => {
+                        this.setState({ isSearching: true, isFetched: false, isValidSearch: true });
+                        this._getComputersBySearch(
+                            this.state.selectedFeild,
+                            this.state.valueToBeSearched
+                        );
+                    }}
+                >
+                    Search!
+                </Button>
+            );
+        }
+    };
+
+    private _getComputersBySearch = async (field: string, value: string) => {
+        const response = await fetch(
+            `/api/${this.context.team.slug}/equipment/GetComputersBySearch?field=${field}&value=${value}`,
+            {
+                method: "GET"
+            }
+        );
+
+        if (!response.ok) {
+            // show the invalid name message.
+            this.setState({ isValidSearch: false });
+        }
+
+        const result = await response.json();
+
+        // if length is 0, not a valid search
+        if (result.length === 0) {
+            this.setState({
+                isFetched: true,
+                isSearching: false,
+                isValidSearch: false
+            });
+        } else {
+
+            // if more than five, shrink the array to only 5 names.
+            if (result.length > 5) {
```

Comment:
You can replace this whole thing with `result.slice(0,5)`.  That'll get the first 5 elements of the array, and if there are less than 5 it'll just return those.  Then this whole area won't need the if statement or for loop, and you don't have to duplicate the setState either